### PR TITLE
Add segment events to accompany speedcurve events

### PR DESCRIPTION
**Pull request by bbaker6225**
Segment events will also likely be useful since we'll probably add
events to the backend as well.

### CODE REVIEW

**Comment by bbaker6225** [comment]
Not in love with these event names, so let me know if you have better ideas

File: src/components/App.jsx [hunk 1]
```diff
@@ -119,9 +120,10 @@ class App extends Component {
   }
 
   render() {
-    // Send performance event on initial render
+    // Send event on initial render
     if (!this.state.hasPreviouslyRendered) {
       utils.markPerformanceIfAble('Payment app began painting');
+      sendTrackEvent('edx.bi.ecommerce.payment_mfe.began_painting');
```
Comment (mulby) [inline src/components/App.jsx]: nit: maybe "started_painting"

File: src/payment/cart/OrderSummary.jsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
 import PropTypes from 'prop-types';
+import { sendTrackEvent } from '@edx/frontend-analytics';
 import { FormattedMessage } from '@edx/frontend-i18n';
 
 import { utils } from '../../common';
 
 class OrderSummary extends React.Component {
   componentDidMount() {
     utils.markPerformanceIfAble('Order Summary component rendered');
+    sendTrackEvent('edx.bi.ecommerce.payment_mfe.order_summary_rendered');
```
Comment (mulby) [inline src/payment/cart/OrderSummary.jsx]: The timestamps on these events are pretty coarse, could we include the milliseconds from the start of the waterfall or something in the event?

Comment (bbaker6225) [reply src/payment/cart/OrderSummary.jsx]: I'll add `performance.now()`. 

`The current high resolution time is the high resolution time from the time origin to the present time (typically called "now").`

I was mainly interested in the included timestamps for getting timing between the backend and frontend. Do you think they have high enough resolution for that?

Comment (mulby) [reply src/payment/cart/OrderSummary.jsx]: I suspect not. Segment makes an HTTP request to their infrastructure when you call `analytics.track()`. The server records the timestamp of the event during the processing of that request. The clock from the client is not trusted.

For this reason, I suspect that two events fired very near one another (milliseconds apart) could actually be processed out-of-order.

Comment (bbaker6225) [reply src/payment/cart/OrderSummary.jsx]: Hmph, ok thanks for the info. I suppose I can just do `Date.now() / 1000` and `time.time()` for the mfe and backend respectively, but that just feels _dirty_

File: src/common/utils.js [hunk 3]
```diff
@@ -136,3 +136,11 @@ export function markPerformanceIfAble(markName) {
     performance.mark(markName);
   }
 }
+
+// Returns performance timing properties to send in analytics requests
+export function getPerformanceProperties() {
+  return {
+    emitted_at: Date.now() / 1000, // Seconds since epoch, for comparing timing on backend vs. frontend
```
Comment (bbaker6225) [inline src/common/utils.js]: I'm not a fan of this but not sure of a better way to compare than using this and `time.time()`

File: src/components/App.jsx [hunk 4]
```diff
@@ -119,9 +120,13 @@ class App extends Component {
   }
 
   render() {
-    // Send performance event on initial render
+    // Send event on initial render
     if (!this.state.hasPreviouslyRendered) {
       utils.markPerformanceIfAble('Payment app began painting');
+      sendTrackEvent(
```
Comment (bbaker6225) [inline src/components/App.jsx]: A concern I have here is that the `identify` call might not have been made yet, in which case we can't have any segment events here

File: src/payment/performanceEventing.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  if (performance.mark) {
+    performance.mark(markName);
+  }
+}
+
+// Returns performance timing properties to send in events
+export function getPerformanceProperties() {
+  return {
+    emitted_at: Date.now() / 1000, // Seconds since epoch, for comparing timing on backend vs. frontend
+    performance_timestamp: performance.now(), // Milliseconds since time origin. See https://developer.mozilla.org/en-US/docs/Web/API/Performance/now
```
[the remark this answers is not in the document]
Comment (mulby) [reply src/payment/performanceEventing.js]: Got it.

**Pull request merged**